pkg/operator/etcdcertsigner/etcdcertsignercontroller.go

```diff
@@ -178,21 +181,60 @@ func (c *EtcdCertSignerController) syncAllMasters(recorder events.Recorder) erro
 	return utilerrors.NewAggregate(errs)
 }
 
+// validateCertificateSAN makes sure that a given certificate is valid
+// at least for the SANs defined in the configuration.
+func (c *EtcdCertSignerController) validateCertificateSAN(certName string, ipAddresses []string) error {
+	secret, err := c.secretLister.Secrets(operatorclient.TargetNamespace).Get(certName)
+	if err != nil {
+		return fmt.Errorf("could not get secrets from %s: %w", certName, err)
+	}
+	if secret.Data == nil {
+		return fmt.Errorf("secret data not found in %s", certName)
+	}
+	tlsCrt, ok := secret.Data["tls.crt"]
+	if !ok {
+		return fmt.Errorf("tls cert data not found in %s", certName)
+	}
+	if len(tlsCrt) == 0 {
+		return fmt.Errorf("%s missing tls.crt content\n", certName)
+	}
+	certificates, err := cert.ParseCertsPEM([]byte(tlsCrt))
+	if err != nil {
+		return fmt.Errorf("could not parse TLS cert for %s: %w", certName, err)
+	}
+	// Using the first certificate to verify IP address match
+	certificate := certificates[0]
+	for _, ipAddress := range ipAddresses {
+		if err := certificate.VerifyHostname(ipAddress); err != nil {
+			return fmt.Errorf("certificate is invalid for %s, %w", ipAddress, err)
+		}
+	}
+	return nil
+}
+
 func (c *EtcdCertSignerController) createSecretForNode(node *corev1.Node, recorder events.Recorder) error {
 	etcdPeerClientCertName := tlshelpers.GetPeerClientSecretNameForNode(node.Name)
 	etcdServingCertName := tlshelpers.GetServingSecretNameForNode(node.Name)
 	metricsServingCertName := tlshelpers.GetServingMetricsSecretNameForNode(node.Name)
 
 	var err error
-	_, err = c.secretLister.Secrets(operatorclient.TargetNamespace).Get(etcdPeerClientCertName)
-	peerClientCertOk := err == nil
-	_, err = c.secretLister.Secrets(operatorclient.TargetNamespace).Get(etcdServingCertName)
-	servingCertOk := err == nil
-	_, err = c.secretLister.Secrets(operatorclient.TargetNamespace).Get(metricsServingCertName)
-	metricsCertOk := err == nil
+
+	nodeInternalIPs, err := dnshelpers.GetInternalIPAddressesForNodeName(node)
+	if err != nil {
+		return err
+	}
+
+	certsOkay := true
+	for _, certName := range []string{etcdPeerClientCertName, etcdServingCertName, metricsServingCertName} {
+		if err := c.validateCertificateSAN(certName, nodeInternalIPs); err != nil {
+			klog.Warningf("validateCertificateSAN: %v", err)
```

Comment:
isnt this an error?

Reply:
We were not printing anything in the past when we regenerated certs :-). But I will make it Errorf.